client/hosting/performance/site-performance.tsx

```diff
@@ -190,6 +190,21 @@ export const SitePerformance = () => {
 		wpcom_performance_report_url?.hash,
 	] );
 
+	useEffect( () => {
```

Comment:
In the spirit of not using `useEffect` 😉 the controller seems like a better place for this to run i.e. the refresh would run "on navigation" so to say. Not "on render".
https://github.com/Automattic/wp-calypso/blob/49f8ef083645a8ff1013bd3bfc2f1498720748b8/client/hosting/performance/controller.tsx#L7-L21

But I see that state like `currentPage` lives as component state, so maybe it's not so easy.

Reply:
Yeah the dependency is on the report being loaded first to determine it's date. We would have to lift the data loading into the navigation, something like [data loading in react query router](https://tanstack.com/router/latest/docs/framework/react/guide/data-loading)

Reply:
This is an interesting point though, would be interesting to see how it works if we start fetching data in the navigation then pass it down to the component.

Reply:
Given the data is fetched using ReactQuery, it should be possible to query that data in a way that's not bound to the React code i.e. using QueryClient to make the query so that both the controller and component are using the same query under the hood rather than firing off two separate requests.

Would be a bit of a refactor though.